Sources/XcodeGenKit/PathKitExtension.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import PathKit
+
+extension Path {
+
+    /// Treat this as a file (resource or source) instead of a normal directory.
+    var isNonFolderDirectory: Bool {
+
+        if isFile || self.extension == nil {
+            return false
+        }
+
+        if let uti = try! URL(fileURLWithPath: self.string)
```

Comment:
I'd like to get the performance tests in (coming shortly) so this can be tested for performance